multi-int-store/src/main/java/com/connexta/multiintstore/config/SolrConfiguration.java

```diff
@@ -18,20 +18,15 @@
 
 @EnableSolrRepositories(basePackages = "com.connexta.multiintstore.repositories")
 @Configuration
-@Profile("default")
 public class SolrConfiguration {
 
-  @NotEmpty
-  @Value("${solr.host}")
-  private String solrHost;
-
-  @Value("${solr.port}")
-  private int solrPort;
-
   @Bean
-  public SolrClient solrClient() {
-    String url = String.format("http://%s:%d/solr", solrHost, solrPort);
-    return new HttpSolrClient.Builder(url).build();
+  @Profile("solrProduction")
```

Comment:
The client is needed regardless of the profile. One itest classes uses a mocked client. Production and the other itest class use the client defined here. There may be a better way to do this. Do you have any suggestions?

Reply:
I realized that I might have misunderstood your question. The S3Client bean does has a profile, just like the SolrClient bean.

Reply:
Looks like if it is on the class, it will apply to all of the beans: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Profile.html. I don't have a preference how we do it here in S3StorageConfiguration and SolrConfiguration.